forgerock-openbanking-uk-aspsp-as/forgerock-openbanking-uk-aspsp-as-gateway/forgerock-openbanking-uk-aspsp-as-gateway-server/src/main/java/com/forgerock/openbanking/aspsp/as/service/registrationrequest/RegistrationRequestJWTClaims.java

```diff
@@ -0,0 +1,149 @@
+     *
+     * @param claimName        @String the name of the SSA claim to obtain from the ssaClaims
+     * @return A non null or empty @String containing the ssa claim value
+     * @throws DynamicClientRegistrationException when the claim specified by the claimName does not exist, or is empty.
+     */
+    public String getRequiredStringClaim(String claimName) throws DynamicClientRegistrationException {
+        Optional<String> claimValue = getOptionalStringClaim(claimName);
+        if(!stringClaimValueIsValid(claimValue)){
+            String errorString = "Claim '" + claimName + "' was not set in " + claimsOrigin + ". This claim " +
+                    "is required";
+            log.debug("getStringClaims() {}", errorString);
+            throw new DynamicClientRegistrationException(errorString, errorType);
+        }
+        return claimValue.get();
+    }
+
+    /**
+     * getRequiredStringClaim Obtains a string containing the value for a specific claim. Catches the
+     * ParseExceptions thrown by nimbusds and instead throw an DynamicClientRegistrationException.
+     *
+     * @param claimName        @String the name of the SSA claim to obtain from the ssaClaims
+     * @return An Optional<String> that will contain the value if it was present or be empty of not
+     * @throws DynamicClientRegistrationException when the claim specified by the claimName does not exist, or is empty.
+     */
+    public Optional<String> getOptionalStringClaim(String claimName) throws DynamicClientRegistrationException {
+        try {
+            return Optional.ofNullable(claimsSet.getStringClaim(claimName));
+        } catch (ParseException e) {
+            String errorString =
+                    "Failed to parse claim '" + claimName + "' from claim set in " + claimsOrigin.toString();
+            log.debug("getStringClaims() {}", errorString, e);
+            throw new DynamicClientRegistrationException(errorString, errorType);
+        }
+    }
+
+    private boolean stringClaimValueIsValid(Optional<String> claimValue){
+        return claimValue.isPresent() && !StringUtils.isEmpty(claimValue.get());
+    }
+
+
+    /**
+     * getRequiredStringListClaims Obtains a list of strings containing the values for a specific claim. Catches the
+     * ParseExceptions thrown by nimbusds and instead throw an DynamicClientRegistrationException.
+     *
+     * @param claimName  {@code String} the name of the SSA claim to obtain from the ssaClaims
+     * @return A non null or empty {@code List<String>} containing the ssa claim value
+     * @throws @DynamicClientRegistrationException when the claim specified by the claimName does not exist, or is
+     * empty.
+     */
+    public List<String> getRequiredStringListClaims(String claimName) throws DynamicClientRegistrationException {
+            Optional<List<String>> values = getOptionalStringListValues(claimName);
+            if (!stringListClaimsAreValid(values)) {
```

Comment:
Shouldnt have to use isPresent and get to check the validity of an optional.
This method could be simplified to
```
return getOptionalStringListValues(claimName)
            .filter(Predicate.not(List::isEmpty))
            .orElseThrow(() -> {
                String errorString = "Claim '" + claimName + "' was not set in " + claimsOrigin + ". This claim " +
                        "is required";
                log.debug("getStringClaims() {}", errorString);
                return new DynamicClientRegistrationException(errorString, getErrorType(claimName));
            });
```

or rename the the stringListClaimsAreValid so it reads better
```
return getOptionalStringListValues(claimName)
            .filter(validClaims())
            .orElseThrow(() -> {
                String errorString = "Claim '" + claimName + "' was not set in " + claimsOrigin + ". This claim " +
                        "is required";
                log.debug("getStringClaims() {}", errorString);
                return new DynamicClientRegistrationException(errorString, getErrorType(claimName));
            });

private Predicate<List<String>> validClaims() {
  return Predicate.not(List::isEmpty);
}
```

Reply:
I like the clarity of using isPresent in some cases, but have taken on your point of using ifPresent where it makes things more readable or the cases are simple.